mesonbuild/backend/ninjabackend.py

```diff
@@ -2595,7 +2602,15 @@ def generate_link(self, target, outname, obj_list, linker, extra_args=None, stdl
         dep_targets.extend([self.get_dependency_filename(t) for t in dependencies])
         dep_targets.extend([self.get_dependency_filename(t)
                             for t in target.link_depends])
-        elem = NinjaBuildElement(self.all_outputs, outname, linker_rule, obj_list, implicit_outs=implicit_outs)
+        # On Windows systems, always use rsp files.
+        # On other systems, only use them for long lines passed to dynamic linker (for now)
+        # FIXME: don't hardcode commandline length limit; do define it better
+        max_len = 1000
+        if linker_base == 'STATIC' or mesonlib.is_windows() or len(obj_list) > max_len:
+            rule = linker_rule
+        else:
+            rule = linker_short_rule
+        elem = NinjaBuildElement(self.all_outputs, outname, rule, obj_list, implicit_outs=implicit_outs)
```

Comment:
Extract the decision logic and its corresponding comment to its own method called `should_use_rsp_file` or something like that.